build-tools/create-dotnet-msi/create-dotnet-msi.csproj

```diff
@@ -0,0 +1,105 @@
+
+  <ItemGroup>
+    <None Remove="**" />
+    <None Include="$(_WixTemplate)" />
+  </ItemGroup>
+
+  <UsingTask AssemblyFile="$(PrepTasksAssembly)"      TaskName="Xamarin.Android.BuildTools.PrepTasks.DownloadUri" />
+  <UsingTask AssemblyFile="$(BootstrapTasksAssembly)" TaskName="Xamarin.Android.Tools.BootstrapTasks.ConvertToRichText" />
+  <UsingTask AssemblyFile="$(BootstrapTasksAssembly)" TaskName="Xamarin.Android.Tools.BootstrapTasks.GenerateWixFile" />
+  <UsingTask AssemblyFile="$(BootstrapTasksAssembly)" TaskName="Xamarin.Android.Tools.BootstrapTasks.UnzipDirectoryChildren" />
+
+  <Target Name="_DownloadWix">
+    <DownloadUri
+        SourceUris="https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip"
+        DestinationFiles="$(AndroidToolchainCacheDirectory)\wix311-binaries.zip"
+    />
+  </Target>
+
+  <Target Name="_UnzipWix"
+      Inputs="$(AndroidToolchainCacheDirectory)\wix311-binaries.zip"
+      Outputs="$(WixToolPath)candle.exe;$(WixToolPath)light.exe">
+    <UnzipDirectoryChildren
+        NoSubdirectory="True"
+        SourceFiles="$(AndroidToolchainCacheDirectory)\wix311-binaries.zip"
+        DestinationFolder="$(WixToolPath)"
+    />
+  </Target>
+
+  <Target Name="_Properties">
+    <ItemGroup>
+      <_Licenses Include="$(DotNetPreviewPath)sdk-manifests\$(DotNetPreviewVersionBand)\Microsoft.NET.Workload.Android\License*" />
+    </ItemGroup>
+    <PropertyGroup>
+      <License Condition=" '$(License)' == '' and '@(_Licenses->Count())' == '1' ">@(_Licenses)</License>
+      <License Condition=" '$(License)' == '' ">$(XamarinAndroidSourcePath)\LICENSE</License>
+      <_LicenseDestination>$(IntermediateOutputPath)LICENSE.rtf</_LicenseDestination>
+      <_WixFile>$(IntermediateOutputPath)Microsoft.NET.Workload.Android.wix</_WixFile>
+      <_WixObj>$(IntermediateOutputPath)Microsoft.NET.Workload.Android.wixobj</_WixObj>
+      <_WixMsi>$(IntermediateOutputPath)Microsoft.NET.Workload.Android.msi</_WixMsi>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="_ConvertLicense"
+      Inputs="$(MSBuildThisFile);$(BootstrapTasksAssembly);$(License)"
+      Outputs="$(_LicenseDestination)">
+    <ConvertToRichText
+        SourceFile="$(License)"
+        DestinationFile="$(_LicenseDestination)"
+    />
+    <ItemGroup>
+      <FileWrites Include="$(_LicenseDestination)" />
+      <None       Include="$(_LicenseDestination)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_GenerateWix"
+      DependsOnTargets="GetXAVersionInfo"
+      Inputs="$(MSBuildThisFile);$(BootstrapTasksAssembly);@(None)"
+      Outputs="$(_WixFile)">
+    <GenerateWixFile
+        Template="$(_WixTemplate)"
+        DestinationFile="$(_WixFile)"
+        DotNetPath="$(DotNetPreviewPath)"
+        DotNetVersion="$(DotNetPreviewVersionFull)"
+        MSIVersion="$(AndroidMSIVersion)"
+    />
+    <ItemGroup>
+      <FileWrites Include="$(_WixFile)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_Candle"
```

Comment:
Not sure if consistency with WiX verbs is "brilliant" or "confusing".  I'm sure these are WiX verbs, but I don't know what they *mean*, so this is decidedly "non-obvious".

Reply:
`_Candle` runs `candle.exe` and `_Light` runs `light.exe`. I don't really know what they are going for with their names.

I could rename these to something like `_Build` and `_Package`? Not sure how much better it is.

Reply:
It's "better" in that someone who knows *nothing* of WiX can still kinda figure out the semantics here.  `_Build` is ???, but `_Package` is "create the desired redistributable."